### PR TITLE
bext: refactor integration tests

### PR DESCRIPTION
**Pull request by taras-yemets**
Closes https://github.com/sourcegraph/sourcegraph/issues/33484

1. Changes browser extension integration tests snapshots compression mechanism. We used `git lfs` to store snapshots before, but used too much bandwidth to fetch them. This PR suggests compressing the snapshots after recording and adding compressed files to repo index. Snapshots are decompressed before and removed after running integration tests.
2. Enables browser extension integration tests in the pipeline as we no longer fetch snapshots with `git-lfs`.
3. Removes browser extension after install page integration test as neither Polly nor Percy support non _http(s)_ protocols.
4. Refactors GitHub search results pages integration tests to omit 429 HTTP errors causing recordings to fail. There's still no guarantee we won't have this error while recording all integration tests at once in the future.

## Test plan
- run `sg test bext-build`
- run `yarn workspace @sourcegraph/browser record-integration` and ensure that 
  - it passes 
  - there are no test recording files (`.har`) in git index
  - compressed recordings (`.har.gz`) are tracked by git  
- run `sg test bext-integration` and ensure that
  - tests pass
  - there are no test recording files (`.har`) in git index

## App preview:
- [Link](https://sg-web-taras-yemets-refactor-bext-tests.onrender.com)

### CODE REVIEW

File: client/browser/scripts/utils.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+const buildDecompressedFilePath = compressedFilePath => compressedFilePath.replace(/\.gz$/, '')
+
+const findRecordingPath = async (path, isCompressed) => {
+  const content = await readdir(path, { withFileTypes: true })
+
+  if (content.length === 0) {
+    return
+  }
+
+  if (content[0].isDirectory()) {
```
Comment (taras-yemets) [inline client/browser/scripts/utils.js:21]: Each snapshot file (_recording.har_) lives in its own directory whose path matches the test full name (see Polly [configuration](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@main/-/blob/client/shared/src/testing/integration/context.ts?L120-123)). 
So we care only about the first item of the array here: if it is a directory, we go deeper the file tree.

File: client/browser/src/integration/gitlab.test.ts [hunk 2]
```diff
@@ -41,6 +41,38 @@ describe('GitLab', () => {
...
+                    })
+                )
+            })
```
Comment (taras-yemets) [inline client/browser/src/integration/gitlab.test.ts:74]: It seems that GitLab changed the way they populate to file content. Mocking it.
image

File: client/browser/scripts/record-integration.js [hunk 3]
```diff
@@ -1,30 +1,32 @@
...
+      `POLLYJS_MODE=record SOURCEGRAPH_BASE_URL=https://sourcegraph.com yarn run-integration --grep='${grepValue}'`,
+      (code, stdout, stderr) => {
+        if (code !== 0) {
+          console.error(code)
+        } else {
+          console.log(`stdout: ${stdout}`)
+          console.error(`stderr: ${stderr}`)
+        }
+
+        resolve()
```
[the remark this answers is not in the document]
Comment (taras-yemets) [reply client/browser/scripts/record-integration.js:19]: The current approach is that we run through all the snapshots and try to compress **all** of them (if found) in each `record-integration` script execution, even if it has been called with the `--grep` argument.

File: client/browser/src/integration/gitlab.test.ts (hunk 2 above)
Comment (tjkandala) [reply client/browser/src/integration/gitlab.test.ts:74]: Would this not be picked up by the recording?

**Review by tjkandala — APPROVED** [review]
Looks great, `utils.js` is so good

File: client/browser/src/integration/gitlab.test.ts (hunk 2 above)
Comment (taras-yemets) [reply client/browser/src/integration/gitlab.test.ts:74]: Nice catch, @tjkandala! Thank you!
Will tackle it in https://github.com/sourcegraph/sourcegraph/pull/33709

**Pull request merged**